wp-includes/sqlite-ast/class-wp-sqlite-driver.php

```diff
@@ -691,6 +727,27 @@ public function query( string $query, $fetch_mode = PDO::FETCH_OBJ, ...$fetch_mo
 		}
 	}
 
+	/**
+	 * Parse a MySQL query into an array of ASTs.
+	 *
+	 * @param  string           $query    The MySQL query to parse.
+	 * @return Generator<WP_Parser_Node>  A generator of ASTs representing the queries parsed from the input.
+	 * @throws WP_SQLite_Driver_Exception When the query parsing fails.
+	 */
+	public function parse_query( string $query ): Generator {
```

Comment:
Do we need a generator? Or could we return the parser here?

Reply:
For now: https://github.com/Automattic/sqlite-database-integration/pull/42/commits/6a3b370b2406defeb5cc6f8ebb0197a31b2caa7c